portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -225,6 +246,8 @@ private Entity createCommentEntity() {
       commentEntity.setProperty(COMMENT_TEXT, this.text);
       commentEntity.setProperty(COMMENT_DATE, this.date);
       commentEntity.setProperty(COMMENT_EMAIL, this.user.emailAddress);
+      commentEntity.setProperty(COMMENT_ID, this.user.id);
+      commentEntity.setProperty(COMMENT_USERNAME, this.user.username);
```

Comment:
I don't think you ever read the username from the commentEntity. I'd argue that it's better to not store the username here at all since 1) you're not currently using it, and 2) you don't have to worry about keeping it in sync when the user changes their username. In fact, i think the username stored with comments may actually fall out of date when a user changes their username multiple times, but it's moot since this copy of the username isn't read.

This logic also applies to the user email to certain degree. Eg: you could imagine a version of this where the comment keeps only a user id and whenever you want to display an email or username with a comment you look up the value you need by id.

Reply:
I check for username updates on every call to the `DataServlet.java`, which means I don't reference it again there, but I read the username on the frontend in order to show the username of the person who posted the comment.
It's possible that doesn't show on your side, but the username and email are both on the frontend of my current version: 
![image](https://user-images.githubusercontent.com/7987279/84837828-dc847680-afed-11ea-9314-b5764e2ec8a2.png)

The email is in a `mailto:` link when you click on the username, so it isn't visible unless you click on the username.

Reply:
I could look it up every time from the ID, but that'd involve calling a servlet again -- since I"m focusing on the capstone, I might leave this for if I have time in the future.

Reply:
ah sorry, I think I was unclear. The point I was trying to articulate is that in getComments you don't read the COMMENT_USERNAME property, instead calling on the AuthServlet to derive the username based on email and id. Which I think means it's not necessary to store the username here. Either way this works correctly from a users perspective.

But acknowledged that you're focused on the capstone. That should definitely be your priority rn, so no need to make changes based on this.